flow/Platform.cpp

```diff
@@ -1932,7 +1933,10 @@ std::string abspath( std::string const& path, bool resolveLinks, bool mustExist
 			if(prefix.empty()) {
 				prefix = ".";
 			}
-			return cleanPath(joinPath(abspath(prefix, true, false), suffix));
+			// Home directory references via ~ are not handled
+			if(prefix[0] != '~') {
```

Comment:
A couple thoughts:

1. My hope is that we wouldn't need to handle this here because realpath would take care of it. Instead, I was thinking this would be more relevant when not resolving links
2. This is a legal directory/file name in Windows, and actually in Linux too for that matter. Additionally, directories like ~foo are possible.

So I think the real question is what we should do with ~ when not resolving links. We could detect that the first path element is ~ and expand it, and that's probably what people would normally expect. It would also make the behavior between resolveLinks being true and false consistent. Or we could instead document that ~ is not resolved when not resolving links and is treated as a normal file or directory name.

Reply:
And for what it's worth, I'm ok with the option chosen here, which is the last one, as long as it's documented. I think we should clean it up though to apply only to unixish environments and to be restricted to a single character path element.

Reply:
Realpath() does NOT take care of ~ references, they're actually a shell feature (of most shells).

I pointed #2 out in a previous comment as well.  I'm OK with ignoring this fact since it's not common to use ~'s in non-temporary files.

I do not agree with restricting it to a single character element.  If we're going to support, even if only to avoid, the home directory notion of ~ then we should support the full syntax equally, to include ~dbUser as a way to refer to the home directory for the user "dbUser".

Reply:
Good to know about realpath. With respect to point 2, I wasn't aware of that syntax and therefore agree that we shouldn't have inconsistent behavior. I still think Windows should ignore the ~ character completely, and in unixish environments I think reverting to the previous behavior of failing may be the option I prefer the most. Or we could just fail in all 3 platforms and not worry about it.

Reply:
Windows PowerShell also supports ~, but I'm not sure about the legacy shell or how often people use the legacy shell.

Also, fdbmonitor uses setuid() so resolving "~" could be inconsistent between the code that constructs a process command line and the launched fdb process. 

I think the current behavior is a good option.  For resolveSymlinks=true, it is impossible to resolve any symlinks if the path starts with ~ so it will (after recursion) fail.  For resolveSymlinks=false, a starting ~[user] component is left alone and not rebased on the current working directory.